src/plugins/action_trim_responses/src/databaseMethods.ts

```diff
@@ -0,0 +1,77 @@
+const databaseMethods = (DBConnect: any) => ({
+  getAllApplicationResponses: async (applicationId: number) => {
+    const text = `
+    SELECT ar.id, code, value, time_created
+    FROM application_response ar JOIN template_element te
+    ON ar.template_element_id = te.id
+    WHERE application_id = $1
+    ORDER BY time_created
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [applicationId] })
+      const responses = result.rows
+      return responses
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  getAllReviewResponses: async (reviewId: number) => {
+    const text = `
+    SELECT rr.id, code, comment, decision, rr.time_created
+    FROM review_response rr JOIN application_response ar
+    ON rr.application_response_id = ar.id
+    JOIN template_element te ON ar.template_element_id = te.id
+    WHERE review_id = $1
+    ORDER BY time_created
+    `
+    try {
+      const result = await DBConnect.query({ text, values: [reviewId] })
+      const responses = result.rows
+      return responses
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  },
+  deleteApplicationResponses: async (responsesToDelete: number[]) => {
+    const deletedCodes: any = []
+    for (const responseId of responsesToDelete) {
+      const text = `DELETE from application_response
+      WHERE id = $1
```

Comment:
Yes, good idea, that makes more sense than a loop. 💡 Will do.

Reply:
Turns out that `node-postgres` doesn't properly support array values for substitutions, although it works with =ANY rather than IN:
`DELETE from review_response WHERE id = ANY ($1)`
Which is basically the same thing, right?

Done in 5d4c0e5